src/cmap/message_stream.ts

```diff
@@ -1,55 +1,91 @@
-const BufferList = require('bl');
-import { Duplex } from 'stream';
-import { Response, Msg, BinMsg } from './commands';
+import BufferList = require('bl');
+import { Duplex, DuplexOptions } from 'stream';
+import {
+  Response,
+  Msg,
+  BinMsg,
+  Query,
+  WriteProtocolMessageType,
+  MessageHeader,
+  CommandResult
+} from './commands';
 import { MongoError, MongoParseError } from '../error';
 import { OP_COMPRESSED, OP_MSG } from './wire_protocol/constants';
 import {
   compress,
   decompress,
-  compressorIDs,
-  uncompressibleCommands
+  uncompressibleCommands,
+  Compressor,
+  CompressorName
 } from './wire_protocol/compression';
+import type { Callback, Document } from '../types';
+import type { ClientSession } from '../sessions';
 
 const MESSAGE_HEADER_SIZE = 16;
 const COMPRESSION_DETAILS_SIZE = 9; // originalOpcode + uncompressedSize, compressorID
 
 const kDefaultMaxBsonMessageSize = 1024 * 1024 * 16 * 4;
 const kBuffer = Symbol('buffer');
 
+interface MessageStreamOptions extends DuplexOptions {
+  maxBsonMessageSize?: number;
+}
+
+export interface OperationDescription {
+  started: number;
+  cb: Callback<CommandResult | undefined>;
```

Comment:
```suggestion
  cb: Callback<CommandResult>;
```

`undefined` is unnecessary since `Callback` has an optional for the result type